src/xmpp_callbacks.ml

```diff
@@ -34,10 +34,14 @@ type user_data = {
 
 let validate_utf8 txt =
   let txt = String.map (fun c -> match c with
-      | '\x00' .. '\x08' |'\x0b'|'\x0c'| '\x0e' .. '\x1f'
-       (* filter <0x20 except if it's tab (x09) or newline (x0a/x0d) *)
-        -> '?'
-      | c -> c
+       (* filter <0x20 except if it's tab (x09) or newline (x0a) *)
+      | '\x09' | '\x0a'
+          -> c
+      | '\x0d' (* CR, turn it into a LF / newline *)
```

Comment:
Re: Double lines: Which will in turn be squashed/eliminated when the text is line wrapped, anyway \r isn't too common in the first place and this fixes the /current/ hack which allowed \r's to go through unfiltered, allowing people to rewrite lines. So, for now, better than not merging it.

Re: Doing all this properly, by replacing the current monster with your library, has already been suggested (by you) and acknowledged as the ideal solution (by hannes): https://github.com/hannesm/jackline/issues/8#issuecomment-67773215
-- Please, go ahead! :-) I think you'd be just the right person to work on this!